modules/product/product.service.js

```diff
@@ -107,14 +106,16 @@ class ProductsService {
     if (await this.checkProductExist(productData, id)) {
       throw new Error(PRODUCT_ALREADY_EXIST);
     }
-    return Product.findByIdAndUpdate(id, productData, { new: true });
+    const updatedProduct = await this.getModel(productData)
+    return Product.findByIdAndUpdate(id, updatedProduct, { new: true });
   }
 
   async addProduct(data) {
     if (await this.checkProductExist(data)) {
       throw new Error(PRODUCT_ALREADY_EXIST);
     }
-    return new Product(data).save();
+    const newProduct = await this.getModel(data)
```

Comment:
It is inconvenient naming, you assign variable called `newProduct` the result of the method that is called `getModel`

Reply:
renamed